packages/init/src/utils/tracker.ts

```diff
@@ -1,57 +1,115 @@
-/*
- * DUPLICATION
- * this file is a duplication of ServerTracker in sm-ui
- * this duplication has been done on purpose to simplify the merge if init becomes part of sm-ui
- * or if we decide to share the tracker implementation from a lib, but it'll need some refactoring
- * Duplication means no entanglement so don't merge this with the other implem and keep the structure.
- */
-
 import ServerAnalytics from "analytics-node";
+import { v4 as uuidv4 } from "uuid";
+import { Frameworks } from "@slicemachine/core/build/src/models";
 
 export enum EventType {
   DownloadLibrary = "SliceMachine Download Library",
+  InitStart = "SliceMachine Init Start",
+  InitDone = "SliceMachine Init Done",
 }
 
-export class Tracker {
-  constructor(
-    readonly analytics: ServerAnalytics,
-    readonly repo: string,
-    readonly identifier: { userId: string } | { anonymousId: string },
-    readonly tracking = true
-  ) {}
-
-  static build(
-    writeKey: string,
-    repo: string | undefined,
-    identifier: { userId: string } | { anonymousId: string },
-    tracking = true
-  ): Tracker | undefined {
+type SegmentIdentifier = { userId: string } | { anonymousId: string };
+
+export class InitTracker {
+  #client: ServerAnalytics | null = null;
+  #isTrackingActive = true;
+  #anonymousId = "";
+  #userId: string | null = null;
+
+  initialize(segmentKey: string, isTrackingActive = true): void {
     try {
-      if (!repo) return;
-      const analytics = new ServerAnalytics(writeKey);
-      return new Tracker(analytics, repo, identifier, tracking);
+      // The initialize method cannot be used for changing some class properties
+      if (!!this.#client) return;
+
+      this.#isTrackingActive = isTrackingActive;
+      this.#anonymousId = uuidv4();
```

Comment:
Note: are we forced to define it ?
Can't the lib take the deviceID by default like the frontend lib does ?